### PR TITLE
Fix parent docker-compose

### PR DESCRIPTION
**Pull request by mhrimaz**
- Fixed monogdb environement variables for username and password in parent [docker-compose](https://github.com/eclipse-basyx/basyx-java-server-sdk/blob/0f526d7462570fae18e85dd22a143a89d59881bb/docker-compose.yml#L7). Previous environment variables are for mongo-express and not mongodb.
- Examples are updated.
- Added an example for running concept description repository.

### CODE REVIEW

**Comment by mhrimaz** [comment]
Wouldn't it make more sense to use testcontainers for mongodb and not use the deployed mongodb from docker-compose.yml parrent?

we already have dependency to testcontainers in aasregistry module.

File: basyx.submodelrepository/basyx.submodelrepository.component/src/main/resources/application.properties [hunk 1]
```diff
@@ -3,12 +3,13 @@ server.port=8081
...
-# basyx.backend = MongoDB
-#spring.data.mongodb.host=localhost
+#basyx.backend = MongoDB
+#spring.data.mongodb.host=mongo
+# or spring.data.mongodb.host=127.0.0.1
 #spring.data.mongodb.port=27017
-#spring.data.mongodb.database=aas
-#spring.data.mongodb.username=admin
-#spring.data.mongodb.password=password
+#spring.data.mongodb.database=admin
```
Comment (alexgordtop) [inline basyx.submodelrepository/basyx.submodelrepository.component/src/main/resources/application.properties:10]: The admin database should be exclusive for mongo configuration -> e.g. authentication database.
The "business data" should be kept in business related databases - like "aas".
See: https://www.mongodb.com/docs/manual/tutorial/manage-users-and-roles/#create-a-user-defined-role

File: examples/cd-repo.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
+server.port=8081
+
+spring.application.name=Concept Description Repository
+
+
+
+basyx.backend = MongoDB
+spring.data.mongodb.host=mongo
+spring.data.mongodb.port=27017
+spring.data.mongodb.database=admin
```
Comment (alexgordtop) [inline examples/cd-repo.properties:10]: See https://github.com/eclipse-basyx/basyx-java-server-sdk/pull/42/files#r1253095824

File: basyx.conceptdescriptionrepository/basyx.conceptdescriptionrepository.component/src/main/resources/application.properties [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
-# basyx.backend = MongoDB
-#spring.data.mongodb.host=localhost
+#basyx.backend = MongoDB
+#spring.data.mongodb.host=mongo
+# or spring.data.mongodb.host=127.0.0.1
 #spring.data.mongodb.port=27017
-#spring.data.mongodb.database=aas
-#spring.data.mongodb.username=admin
-#spring.data.mongodb.password=password
+#spring.data.mongodb.database=admin
```
Comment (alexgordtop) [inline basyx.conceptdescriptionrepository/basyx.conceptdescriptionrepository.component/src/main/resources/application.properties:10]: https://github.com/eclipse-basyx/basyx-java-server-sdk/pull/42/files#r1253095824

File: basyx.aasenvironment/basyx.aasenvironment.component/src/main/resources/application.properties [hunk 4]
```diff
@@ -3,6 +3,14 @@ spring.application.name=AAS Environment
 
 basyx.backend = InMemory
 
+#basyx.backend = MongoDB
+#spring.data.mongodb.host=mongo
+# or spring.data.mongodb.host=127.0.0.1
+#spring.data.mongodb.port=27017
+#spring.data.mongodb.database=admin
```
Comment (alexgordtop) [inline basyx.aasenvironment/basyx.aasenvironment.component/src/main/resources/application.properties:10]: https://github.com/eclipse-basyx/basyx-java-server-sdk/pull/42/files#r1253095824

**Comment by alexgordtop** [comment]
> 

@mhrimaz The issue with test containers is, that in an multiarch environment, it's hard to use them (some develop on amd64, some on arm64). Additionally, as soon as a replicaset is required (for e.g. acid transactions on mongo), test containers do not offer this option - or at least did not in the past, didn't double check that again.

**Comment by mhrimaz** [comment]
> > 
> 
> @mhrimaz The issue with test containers is, that in an multiarch environment, it's hard to use them (some develop on amd64, some on arm64). Additionally, as soon as a replicaset is required (for e.g. acid transactions on mongo), test containers do not offer this option - or at least did not in the past, didn't double check that again.
> 
> And another important aspect is, that we can look at the resulting database content after the tests. The test containers are clean up, and it's much more complicated to look at the results between two runs.


@alexgordtop Thanks a lot for clarification and explanation. However, a change in the mongodb port, url, ... will cause issues and you need to manually replace the connection string in many places, which I guess it isn't nice (Maybe I am wrong?).

Anyway, regarding the other reviews, I will solve them and get back again.

Thanks

**Comment by mhrimaz** [comment]
@alexgordtop 
Parameter `spring.data.mongodb.database` should be used to name the business db, as you mentioned. However, when you have authentication for db, you need to explicitly define the authentication db with  `spring.data.mongodb.authentication-database` parameter.

We can also increase the mongodb version to version 6, which has better query engine performance. What do you think?

**Comment by FrankSchnicke** [comment]
Thank you very much for your valuable contribution!

**Pull request merged**